src/lib/y2users/linux/writer.rb

```diff
@@ -101,8 +101,15 @@ def initialize(config, initial_config)
       def write
         issues = Y2Issues::List.new
 
+        # handle groups first as it does not depend on users uid, but it is vice versa
+        new_groups = config.groups.map(&:id) - initial_config.groups.map(&:id)
+        new_groups.map! { |id| config.groups.find { |g| g.id == id } }
+        new_groups.each { |g| add_group(g, issues) }
+
+        # TODO: modify group?
+
         users_finder.added.each { |u| add_user(u, issues) }
-        users_finder.password_edited.each { |u| change_password(u, issues) }
+        users_finder.modified.each { |nu, ou| modify(nu, ou, issues) }
```

Comment:
NP: I'd write `new_user` and `old_user`

Reply:
well, my internal rule ( and I think also rubocop has it ) if it is `{}` block so one liner, short names can be used. If it is longer one, I use more descriptive name.